UploadDaemon/SymbolAnalysis/SymbolCollectionResolver.cs

```diff
@@ -41,19 +52,32 @@ private static (string, GlobPatternList) GetCacheKey(string symbolDirectory, Glo
             return (symbolDirectory, assemblyPatterns);
         }
 
-        private bool IsValid(SymbolCollection collection)
+        private bool IsValid(SymbolCollection collection, ICollection<SymbolFileInfo> relevantSymbolFiles)
         {
-            return collection.SymbolFileNames.All(symbolFileName => ReadLastWriteDate(symbolFileName).Equals(symbolFileLastWriteDates[symbolFileName]));
+            return ContainsExactly(collection, relevantSymbolFiles)
+                && relevantSymbolFiles.All(symbolFile => symbolFile.LastWriteTime.Equals(symbolFileLastWriteDates[symbolFile.Path]));
         }
 
-        private void UpdateValidationInfo(SymbolCollection collection)
+        private static bool ContainsExactly(SymbolCollection collection, ICollection<SymbolFileInfo> relevantSymbolFilePaths)
         {
-            foreach (string symbolFileName in collection.SymbolFileNames)
+            return new HashSet<string>(relevantSymbolFilePaths.Select(info => info.Path)).SetEquals(collection.SymbolFilePaths);
```

Comment:
Minor: AFAICS, symbolfilepaths is only used for this check. Could we make that a set, so that we don't need to create a hashset here every time? would make the code easier to read and also make it clearer what the purpose of the SymbolFilePaths set is :-)